src/Sentry/SentrySdk.cs

```diff
@@ -78,13 +71,16 @@ public static IDisposable Init(Action<SentryOptions>? configureOptions)
         [EditorBrowsable(EditorBrowsableState.Never)]
         public static IDisposable Init(SentryOptions options)
         {
-            if (options.Dsn == null)
+            if (string.IsNullOrWhiteSpace(options.Dsn))
             {
-                if (!Dsn.TryParse(DsnLocator.FindDsnStringOrDisable(), out var dsn))
+                var dsn = DsnLocator.FindDsnStringOrDisable();
+
+                if (Dsn.TryParse(dsn) is null)
```

Comment:
if it's not `null` (which means it could locate any DSN), and it's not `""` (which is intentional "disable SDK please", according to sentry SDK spec) and it failed parsing, it's OK to throw an exception here.

It means the user has a broken DSN and it's worth letting them know, since it'll fail consistently here during app bootstrap.